service_crategen/src/main.rs

```diff
@@ -0,0 +1,287 @@
+#[macro_use]
+extern crate clap;
+extern crate rayon;
+extern crate rusoto_codegen;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde_json;
+extern crate toml;
+
+mod cargo;
+
+use std::collections::BTreeMap;
+use std::fs::{self, File, OpenOptions};
+use std::io::{Read, Write, BufWriter};
+use std::path::Path;
+
+use rayon::prelude::*;
+
+use clap::{Arg, ArgGroup, App};
+
+use rusoto_codegen::botocore::Service;
+
+#[derive(Deserialize)]
+struct ServiceConfig {
+    pub version: String
+}
+
+fn get_dependencies(service: &Service, core_version: &str) -> BTreeMap<String, cargo::Dependency> {
+    let mut dependencies = BTreeMap::new();
+
+    dependencies.insert("hyper".to_owned(), cargo::Dependency::Simple("0.10.0".into()));
+    dependencies.insert("rusoto_core".to_owned(), cargo::Dependency::Extended {
+        path: Some("../../core".into()),
+        version: Some(core_version.to_owned()),
+        optional: None,
+        default_features: None,
+        features: None
+    });
+
+    match &service.metadata.protocol[..] {
+        "json" => {
+            dependencies.insert("serde".to_owned(), cargo::Dependency::Simple("0.9.5".into()));
```

Comment:
I do like the idea of building this cargo manifest with code.  I don't like keeping these lists of dependencies and their versions as "magic numbers" in the code.  Making a rust code change to update a `serde` dependency version seems counter-intuitive, even if it's a dependency in a crate generated by this executable.  

Not a reason to stop this merge, but we should open an issue for making this code read some kind of configuration file and do the exact same thing it's doing now with that configuration.